### PR TITLE
Aeh/more icons

### PR DESCRIPTION
**Pull request by aidan-hubley**
**Changes**
- Added custom post type icons
- Added icons to circle icon selector
- Icon selector component is now a Flatlist
- Icon selector icons are now categorized and separated by headers

**Notes**
N/A

**YouTrack Link**
[P149](https://prayercircle.youtrack.cloud/agiles/147-2/current?issue=PC-149)

### CODE REVIEW

**Review by dev-arekusandoru — APPROVED** [review]
Looks good. Thanks for taking the time to do this.

File: app/(aux)/detailedJournal.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default function Page() {
 						timestamp={1695846631107}
 						content='He is very sick blah blah blah blah oh blah blah blah blah blah blah blah
 blah blah blah blah blah blah blah oh no he’s gonna die ahhhhhhhhhhh'
-						icon='heart-outline'
+						// icon='request'
```
Comment (dev-arekusandoru) [inline app/(aux)/detailedJournal.js]: these being commented out causes and `icon type undefined_outline` error

File: app/(main)/profile.js [hunk 2]
```diff
@@ -177,7 +177,7 @@ export default function ProfilePage() {
 						title={item[1].title}
 						timestamp={`${item[1].timestamp}`}
 						content={item[1].text}
-						icon='heart-outline'
+						icon='prayer'
```
Comment (dev-arekusandoru) [inline app/(main)/profile.js]: this should be updated to `item[1].type` rather than a static value

File: components/Post.js [hunk 3]
```diff
@@ -77,11 +78,55 @@ export const Post = (post) => {
...
+		event: {
+			outline: require('../assets/post/calendar_outline.png'),
+			nonOutline: require('../assets/post/calendar.png'),
+		},
+		request: {
+			outline: require('../assets/post/prayer_outline.png'),
+			nonOutline: require('../assets/post/prayer.png'),
+		},
+		prayer: {
+			outline: require('../assets/post/prayer_outline.png'), // replace with the correct image paths
```
Comment (dev-arekusandoru) [inline components/Post.js]: what does the comment here mean?

File: components/Post.js [hunk 4]
```diff
@@ -77,11 +78,55 @@ export const Post = (post) => {
...
 	function toggleIcon() {
-		if (icon.includes('-outline')) {
-			setIcon(icon.replace('-outline', ''));
-		} else {
-			setIcon(icon + '-outline');
+		const iconKey = iconType.replace('_outline', '');
+		if (['praise', 'event', 'request', 'prayer'].includes(iconKey)) {
+			setIconType(iconType.includes('outline') ? iconKey : iconKey + '_outline');
+			
+			Animated.sequence([
```
Comment (dev-arekusandoru) [inline components/Post.js:118]: I think this animation is a good idea, but not entirely necessary. a simpler approach would be to have the iconAnimation go from 0-1 and then have the input/output values something like 
`input: [0, 0.5, 1],
output: [1, 1.5, 1]`
and this way you can just call the Animated timing 1 time that goes to 0 if it is 1 and vice versa. also icon change is really delayed when the animation plays, not sure if you have seen/thought about that

File: components/iconSelector.js [hunk 5]
```diff
@@ -25,108 +25,221 @@ const StyledPressable = styled(Animated.createAnimatedComponent(Pressable));
...
+            />
+            <StyledAnimView
+                style={{ opacity: opacityInterpolation }}
+                pointerEvents={opened ? 'auto' : 'none'}
+                className="absolute -translate-x-[150px] left-1/2 top-[60%] -translate-y-[280px] w-[80%] p-[15px] max-w-[300px] h-[85%] max-h-[500px] bg-offblack border border-[#3D3D3D] rounded-[20px]"
+            >
+                <StyledText className="text-offwhite font-bold text-3xl text-center mb-3">
+                    Select an Icon
+                </StyledText>
+                <StyledScrollView>
```
Comment (dev-arekusandoru) [inline components/iconSelector.js]: The load times are really slow. This scroll view should be changed out with a FlatList

**Review by dev-arekusandoru — COMMENTED** [review]
A few more comments before this is ready for merge

File: components/Post.js (hunk 4 above)
Comment (aidan-hubley) [reply components/Post.js:118]: I think I'd like to compare the differences on our builds. I wouldn't say the icon change is delayed on mine.

File: components/iconSelector.js (hunk 5 above)
Comment (aidan-hubley) [reply components/iconSelector.js]: Mine load immediately, I will test changing

File: components/Post.js (hunk 4 above)
Comment (dev-arekusandoru) [reply components/Post.js:118]: sure. send a video on slack ill send one as well. even if it isn't delayed on yours we'll need to account for all devices

Comment (aidan-hubley) [reply components/Post.js:118]: I would consider this functional, and if you would like to open a new pr to resolve this 'bug' go for it

**Pull request merged**